pushHandler.go

```diff
@@ -37,20 +37,24 @@ func (h *pushHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	push := payload.(github.PushPayload)
 	downloadURL := getHTTPDownloadURL(push)
 
+	/*
+		Get random tar name
+		This is done early to have a unique temporary directory
+	*/
+	tarName := guuid.New().String()
+
 	// Get Repo Contents
 	Info.Println("Downloading Repo...")
+	tempRepoPath := "./" + tarName
```

Comment:
Nit: might want to prefix it with `git` or smth?
```suggestion
	tempRepoPath := "./git_" + tarName
```